cli/src/complete.rs

```diff
@@ -159,6 +159,37 @@ pub fn aliases() -> Vec<CompletionCandidate> {
     })
 }
 
+pub fn workspaces() -> Vec<CompletionCandidate> {
+    with_jj(|mut jj, _| {
+        let output = jj
+            .arg("workspace")
+            .arg("list")
+            .output()
+            .map_err(user_error)?;
+        let stdout = String::from_utf8_lossy(&output.stdout);
+
+        // example 'workspace list' output for reference:
+        //
+        // default: qpzoy 3ea66ab9 (no description set)
+        // test-ws: powxr b19e946c (empty) (no description set)
```

Comment:
`jj workspace list` uses the `templates.commit_summary` template to render the workspace's current working copy commit. This means that the output format listed here might not be the actual output of `jj workspace list` if the user has modified the commit summary template.

A possible workaround is to specify `jj --config-toml templates.commit_summary "[custom template]"` or add a `jj workspace list --template "[custom template]" argument to ensure that the output format matches what we expect.

Reply:
Maybe we can use `jj log -r 'working_copies()' -T..`?

Reply:
I applied the `--config-toml` approach and it seems to work well. Is there an advantage to using `log`? I think it could complicate some cases, for example when two workspaces point at the same revision.

Reply:
No. I just meant `log` can be used as a tool to query this kind of things. (And I wouldn't want to add `workspace list -T` that only affects the summary part.)